lib/pbench/agent/results.py

```diff
@@ -161,7 +39,7 @@ def __init__(
         self.tarball = Path(tarball)
         if not self.tarball.exists():
             raise FileNotFoundError(f"Tarball '{self.tarball}' does not exist")
-        server_rest_url = config.get("results", "server_rest_url")
+        server_rest_url = config.get("put-server", "server_rest_url")
```

Comment:
This module is in the `agent` side; if we're going to share between server and agent, it should be in `common`. And that suggests devising a new config setting that will be the same on both. While the `server_rest_url` wasn't too bad for the agent side where historically the only server interaction was when uploading tarballs, we already have `pbench-token-generate`.

I think we should probably have a full URI _prefix_ (`http://10.1.170.246/api/v1`) in the `pbench-server` config section, on both server and agent, which would be referenced here, with the endpoints added. (Either hardcoded or with specific endpoint suffix config items.)

Reply:
Yeah, I was planning to move it on the server side or common but this eases the review so I left it as it is.

Reply:
Once Peter's code will be merged I will use that in place of the existing

Reply:
I wanted to keep the put-shim config file separate that's why I created another config file. but if keeping it in the same file doesn't make much of a change than I will change that